recipes/webhooks_example.ipynb

```diff
@@ -189,52 +189,110 @@
+    "    width = max(x_array) - min(x_array)\n",
+    "    height = max(y_array) - min(y_array)\n",
+    "    return width * height\n",
+    "\n",
     "\n",
     "class PluginHandler(PluginCore):\n",
     "    \"\"\"\n",
     "    Custom plugin instance\n",
     "    \"\"\"\n",
     "\n",
+    "    def check_complex_rules_on_label(self, asset_id: str):\n",
+    "        \"\"\"\n",
+    "        Custom method to check if a box is larger than 33% of the image\n",
+    "        For basic rules, a handy object is `search` that \\\n",
+    "            provide various analytics on objects\n",
+    "        In this more complex use-case, we will \\\n",
+    "            fetch the complete `jsonResponse`\n",
+    "        \"\"\"\n",
+    "        json_response = self.kili.labels(\n",
+    "            asset_id=asset_id,\n",
+    "            project_id=self.project_id,\n",
+    "            fields=['jsonResponse'],\n",
+    "            disable_tqdm=True\n",
+    "        )[0]['jsonResponse']\n",
+    "\n",
+    "        issues_array = []\n",
+    "        for annotation in json_response['JOB_0']['annotations']:\n",
+    "            bounding_box = annotation['boundingPoly'][0]['normalizedVertices']\n",
+    "            area = _get_area(bounding_box)\n",
+    "            # Refuse bounding boxes larger than 0.33\n",
+    "            if area > 0.33:\n",
+    "                issues_array.append({\n",
+    "                    'text': 'BBox too large',\n",
+    "                    'mid': annotation[\"mid\"]\n",
+    "                })\n",
+    "\n",
+    "        return issues_array\n",
+    "\n",
     "    def on_submit(self, label: Dict, asset_id: str) -> None:\n",
     "        \"\"\"\n",
-    "        Dedicated handler for Submit action \n",
+    "        Dedicated handler for Submit action\n",
     "        \"\"\"\n",
     "        self.logger.info(\"On submit called\")\n",
     "\n",
     "        issues_array = check_rules_on_label(label)\n",
     "\n",
+    "        issues_array += self.check_complex_rules_on_label(asset_id)\n",
+    "\n",
     "        project_id = self.project_id\n",
     "\n",
     "        if len(issues_array) > 0:\n",
-    "            print(\"Creating an issue...\")\n",
+    "            print(f'Creating {len(issues_array)} issues...')\n",
     "\n",
-    "            for i, _ in enumerate(issues_array):\n",
+    "            for issue in issues_array:\n",
+    "                print(issue)\n",
     "\n",
-    "                self.kili.append_to_issues(\n",
-    "                    label_id=label['id'],\n",
+    "                self.kili.create_issues(\n",
+    "                    label_id_array=[label['id']],\n",
     "                    project_id=project_id,\n",
-    "                    text=issues_array[i],\n",
+    "                    text_array=[issue['text']],\n",
+    "                    object_mid_array=[issue['mid']]\n",
     "                )\n",
```

Comment:
Why not create all the issues at once in batch ?

Reply:
Not possible, misconception, the batch only works for one issue per asset

Reply:
Hmm, okay